activerecord/lib/active_record/associations/builder/has_one.rb

```diff
@@ -26,5 +32,33 @@ def self.define_validations(model, reflection)
         model.validates_presence_of reflection.name, message: :required
       end
     end
+
+    def self.touch_record(o, changes, name, touch, touch_method) # :nodoc:
+      record = o.send name
+
+      return unless record && record.persisted?
+
+      if touch
+        record.send(touch_method)
+      else
+        record.send(touch_method, touch)
```

Comment:
Why `record.send(touch_method, touch)` when `touch` is falsy?

Reply:
Thanks, @kamipo . This was not needed. I have removed that and tested in an app.